src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AddUsersStep.ts

```diff
@@ -0,0 +1,167 @@
+import pluralize from 'pluralize'
+import { ReactNode, useEffect, useState } from 'react'
+import { div, h, p } from 'react-hyperscript-helpers'
+import { useUniqueId } from 'src/components/common'
+import { ValidatedInput } from 'src/components/input'
+import { formHint, FormLabel } from 'src/libs/forms'
+import * as Utils from 'src/libs/utils'
+import {
+  columnEntryStyle,
+  columnStyle, hintStyle,
+  rowStyle
+} from 'src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/styles'
+import {
+  LabeledRadioButton, LabeledRadioGroup
+} from 'src/pages/billing/NewBillingProjectWizard/StepWizard/LabeledRadioButton'
+import { Step } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/Step'
+import {
+  legendDetailsStyle,
+  StepFieldLegend,
+  StepFields
+} from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepFields'
+import { StepHeader } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepHeader'
+import { validate } from 'validate.js'
+
+
+interface EmailInputProps {
+  emails: string
+  onSetEmails: (emails: string, hasError: boolean) => void
+  disabled: boolean
+  errors: ReactNode
+  setErrors: (ReactNode) => void
+  onFocus: () => void
+  label: string
+  addSpacing: boolean
+  inputDebounce?: number
+}
+
+const EmailInput = (props: EmailInputProps) => {
+  const inputId = useUniqueId()
+  const [debouncedEmails, setDebouncedEmails] = useState<ReactNode>()
+  const [debouncedErrors, setDebouncedErrors] = useState<ReactNode>()
+
+  useEffect(() => {
```

Comment:
I think we could make a debounced version of the `ValidatedInput` input, for wider consumption